### DISCUSSION TITLE
Applying background opacity to a gradient background

### DISCUSSION THREAD

**Discussion by gfungx** [tailwindlabs/tailwindcss#3433] [Help]
It seems that I am unable to apply any `bg-opacity-{value}` classes to an element that has any form of background gradient. Currently I am trying to do this here:

```js
<div className="w-full bg-gradient-to-r from-purple-400 via-pink-500 to-red-500 bg-opacity-25 rounded p-6 mt-4">
        // children...
</div>
```

If you were to run this code, you would just see a block with a full opacity gradient in the background, without the `bg-opacity-25` applying. Are there any work arounds or solutions for this?

**Comment by hugovntr** [comment]
Hello,

I often find myself in the same scenario so I found a workaround to get the effect you want. Basically, since the `.bg-opacity` has no effects on gradients _(as well as CSS variables)_ we can use the regular `.opacity` combined with some `.absolute` to get the effect to work.

```html
<div class="relative w-full">
    <div class="absolute inset-0 bg-gradient from-color to-color opacity-25"></div>
    <div class="relative">
        <!-- children -->
    </div>
</div>
```

**Reply by rushkii** [reply]
it depends on the case, if there is a text in the children and the text will apply the opacity too.

**Comment by marcia-lima** [comment]
The following discussions might be related to yours:

- [#2429](https://github.com/tailwindlabs/tailwindcss/discussions/2429)
- [#3617 ](https://github.com/tailwindlabs/tailwindcss/discussions/3617)
- [#3425 ](https://github.com/tailwindlabs/tailwindcss/discussions/3425)

**Comment by cosminepureanu** [comment]
Using Tailwind v3.0
Nowadays you can add `/25` to the color to make it 25% opaque.
In your case, this should work:

```` HTML
<div className="w-full bg-gradient-to-r from-purple-400/25 via-pink-500/25 to-red-500/25 rounded p-6 mt-4">
        // children...
</div>
````

**Reply by jtarallo** [reply]
Could it be this doesn't work if the colors are custom being extended from the them in tailwind's cfg? If it does and you happen to know how, it'd be of help. Ty!

**Reply by NovusOpus** [reply]
@jtarallo Yes, you can use it with extended custom colors, here's an example: [Tailwind Play](https://play.tailwindcss.com/I1VwamN4a5)

**Reply by jtarallo** [reply]
Ty @NovusOpus, I've checked and turns out we are using tailwind 2.2 so we can't use it :(
Thanks for the info though.

**Reply by NovusOpus** [reply]
@jtarallo You can still use it with 2.2, but you have to enable [JIT Mode](https://v2.tailwindcss.com/docs/just-in-time-mode#enabling-jit-mode)

**Reply by jtarallo** [reply]
@NovusOpus  that's really helpful! thanks!

**Reply by jonathanMelly** [reply]
thanks a lot for sharing this info

**Reply by Muffinman** [reply]
This is great, but it would be even better if in the docs! 🥲

**Reply by Standard-IO** [reply]
> This is great, but it would be even better if in the docs! 🥲

It is in the sections background color and upgrade guide. But must be at least a note in the gradient section because is a special case.

**Reply by aun-shahbaz-awan** [reply]
Work Fine!

**Reply by DagNo1** [reply]
Thanks a lot

**Reply by about-hector** [reply]
This doesn't work when using custom colors like from-[var(--starting-color)]/20 etc

**Reply by ghost** [reply]
Thanks

**Reply by is-it-ayush** [reply]
This works if you have a color hard-coded directly into the config although would fail if you're using css variable colors such as,
```ts
colors: {
  red: "var(--color-red)",
}
```
Here's a [quick repro](https://play.tailwindcss.com/5nTWybIRqX).

**Reply by depsimon** [reply]
You need to use a rgb/hsl notation for your colors. Here's your repro fixed. https://play.tailwindcss.com/3J4X83cxrk?file=css

**Reply by is-it-ayush** [reply]
Yep! I did figure this out later but forgot to update it here. Thank you @depsimon!!

**Reply by kavinda1995** [reply]
YOU SAVED ME sensei!
